x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/assembler/ExecutionManager.java

```diff
@@ -65,13 +68,30 @@ public Executable assemble(List<List<Attribute>> listOfKeys,
         for (int i = 0; i < plans.size(); i++) {
             List<Attribute> keys = listOfKeys.get(i);
             List<HitExtractor> keyExtractors = hitExtractors(keys, extractorRegistry);
+            List<String> keyFields = new ArrayList<>(keyExtractors.size());
+
+            // extract top-level fields used as keys to optimize query lookups
+            // this process gets skipped for nested fields
```

Comment:
Why is this skipped for nested fields? It's because of the complexity added by creating a `nested` query?

Reply:
We don't support nested fields in general since they are a 1:n relationship. The check is currently performed in `createExtractor` (and really should be moved into the Verifier instead).